internal/pkg/repository/repository.go

```diff
@@ -0,0 +1,74 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package repository provides support for building and pushing images to a repository.
+package repository
+
+import (
+    "fmt"
+    "github.com/aws/copilot-cli/internal/pkg/aws/ecr"
+)
+
+// DockerService provides support for logging in to repositories, building images and pushing images to repositories.
+type ContainerManager interface {
+    Build(uri, path, imageTag string, additionalTags ...string) error
+    Login(uri, username, password string) error
+    Push(uri, imageTag string, additionalTags ...string) error
+}
+
+// RepositoryGetter gets information of ECR repositories.
+type ECRRepositoryGetter interface {
+    GetRepository(name string) (string, error)
+    GetECRAuth() (ecr.Auth, error)
```

Comment:
I'm thinking to make this pkg more flexible in the long run with other registries we should change the function signature for the `ecr` method to be:
`Auth() (username, password string, error)` using [Go's named return values](https://tour.golang.org/basics/7) this way this interface can be:

```go
type Registry interface {
   RepositoryURI(name string) (string, error)
   Auth() (username, password string, error)
}
```

Reply:
We can do the `ecr` update in a separate PR, when refactoring the existing code to use this new package :)

Reply:
yay agreed. I went back and forth with the decision of whether to add the "ECR" prefix, and the only reason why I finally added "ECR" is because of the method's name `GetECRAuth`. I wasn't sure if it was okay to rename `GetECRAuth` to `Auth`. It's great that you are confirming this!